### ISSUE TITLE
Request: AMD HIP support

### ISSUE THREAD

**Issue by oscarbg** [SveSop/nvidia-libs#3]
Hi,
great project!
just tested VisRTX 0.3.1 Windows app (with ANARI support being Optix based) and worked great!

but anyway my request:
just first AMD HIP Windows big app is here with Blender 3.x.. currently has RNDA+ support but in Blender 3.2 builds we have Vega+ (gfx9+) support..
Cycle support for HIP on Linux builds is coming soon with first builds based on (https://developer.blender.org/D14360) here.(https://builder.blender.org/download/patch/blender-3.2.0-alpha+master-D14360.93d4d10c0432-linux.x86_64-release.tar.xz)
but they are RDNA+ at the moment..
so a simple motivation, is I have a Vega, and I would love to test Blender 3.2 Win builds on Linux.. but for HIP support require Wine HIP support..
basically similar to nvcuda wine support as amdhip64.dll entrypoints are very similar to nvcuda (nv cuda driver calls) just changing cu for hip like cuCreate.. for hipCreate.. ..
are you interested on this or better ask Wine devs directly or wine-staging devs.. in that case can you point me where to ask..
thanks..

**Comment by SveSop** [comment]
Hmm.. I don't know much about this HIP thingy, but some searches around the web kinda seems to me this is a layer that would use cuda if you have an NVIDIA adapter and AMD ROCm (whatever that is) if you have AMD?

Would Blender load nvcuda.dll if it detects an NVIDIA adapter, and amdhip64.dll if it detects an AMD adapter? I kinda fear it is a bit more too it than just relaying calls from amdhip64.dll <-> libamdhip64.so, but at this time i am clueless :smirk: 

So when you install an AMD driver in Linux you will have libamdhip64.so, or is that part of some other package like ROCm?

Sorry for the stupid questions, but this is the first time i hear about HIP.

As a note for wine dev's, they have said they are not really interested in implementing vendor specific libraries like nvcuda, and this is mostly (i believe) the reason it has not moved out of staging. If amdhip64.dll would work similar to how nvcuda.dll works today, i fear this might be a uphill battle to get implemented. If you want to get a reply i would suggest the following:

1. Find some relevant documentation about HIP in general. (Promising that my initial perception of this was that it was vendor neutral - like OpenCL perhaps).
2. Explain what this amdhip64.dll implementation is supposed to achieve (Perhaps not the best example to run Blender with HIP support, since Blender is also available as a native Linux app).
3. Post this as a RFC (Request For Comments) in the wine-devel mailing list : [https://www.winehq.org/forums](https://www.winehq.org/forums) You need to subscribe and then send a mail to the mailinglist. This way the wine developers can give you their thoughts. I have no idea how to reach them other than that.. well.. some dev's hang around on the Linux Gaming Dev discord channels at times tho, but not sure that is the best way to reach them.

I would not really take on such a task without any AMD hardware for testing, especially since i am in no way a real programmer... i am just mediocre at using copy-paste with some flavors :rofl:

**Comment by SveSop** [comment]
Closing this as something i have no plans on implementing any time soon

**Comment by mirh** [comment]
https://github.com/vosen/ZLUDA/issues/72
